thorlcr/graph/thgraphslave.cpp

```diff
@@ -140,10 +140,31 @@ void CSlaveActivity::setInput(unsigned index, CActivityBase *inputActivity, unsi
     else
         outLink.set(((CSlaveActivity *)inputActivity)->queryOutput(inputOutIdx));
     assertex(outLink);
+
     while (inputs.ordinality()<=index) inputs.append(NULL);
+
     inputs.replace(outLink.getClear(), index);
 }
 
+void CSlaveActivity::appendOutput(IThorDataLink *itdl)
+{
+    if (queryJob().getOptBool("TRACEROWS"))
+    {
+        const unsigned numTraceRows = queryJob().getOptInt("numTraceRows", 10);
+        Owned<IThorDataLink> newItdl;
+        newItdl.setown(new CTracingThorDataLink(itdl, queryHelper(), numTraceRows));
+        outputs.append(newItdl.getClear());
```

Comment:
minor: there was a previous comment here, seems to have got lost.
Comment was something along the lines of : Not a lot point in the newItdl here, clearer if above 3 lines replaced with:
```
outputs.append(new CTracingThorDataLink(itdl, queryHelper(), numTraceRows());
```